### ISSUE TITLE
[no-unused-expressions] False positive for conditional chaining function calls

### ISSUE THREAD

**Issue by hisuwh** [typescript-eslint/typescript-eslint#1241]
**Repro**

```JSON
{
  "rules": {
    "no-unused-expressions": "error"
  }
}
```

```ts
someFunction?.();
```

**Expected Result**

I would expect this to be valid because I only want to call the function if its not null.

**Actual Result**

Getting a no-unused-expressions error

**Additional Info**

N/A

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.7.0` |
| `@typescript-eslint/parser`        | `2.7.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `10.9.0` |
| `npm`                              | `6.2.0` |

**Comment by a-tarasyuk** [comment]
I think you need to use `@typescript-eslint/no-unused-expressions` instead of `no-unused-expressions`.

cc @bradzacher

**Comment by bradzacher** [comment]
Thanks @a-tarasyuk .

Yup @hisuwh - make sure you use our extension rules which add support for typescript syntax.

**Comment by hisuwh** [comment]
Of course.  Thanks
